configure.ac

```diff
@@ -257,14 +257,21 @@ AC_ARG_WITH(builtin-grub2-mkconfig,
               [with_builtin_grub2_mkconfig=no])
 AM_CONDITIONAL(BUILDOPT_BUILTIN_GRUB2_MKCONFIG, test x$with_builtin_grub2_mkconfig = xyes)
 AM_COND_IF(BUILDOPT_BUILTIN_GRUB2_MKCONFIG,
-           AC_DEFINE([USE_BUILTIN_GRUB2_MKCONFIG], 1, [Define if using internal ostree-grub-generator]),
-[
-  # Otherwise, look for the path to the system generator.  On some
-  # distributions GRUB2 *-mkconfig executable has 'grub2' prefix and
-  # on some 'grub'.
-  AC_CHECK_PROG(GRUB2_MKCONFIG, grub2-mkconfig, grub2-mkconfig, grub-mkconfig)
-  AC_DEFINE_UNQUOTED([GRUB2_MKCONFIG_PATH], ["$GRUB2_MKCONFIG"], [The system grub2-mkconfig executible name])
-])
+           AC_DEFINE([USE_BUILTIN_GRUB2_MKCONFIG], 1, [Define if using internal ostree-grub-generator]))
+AC_ARG_WITH(grub2-mkconfig-path,
+            AS_HELP_STRING([--with-grub2-mkconfig-path],
+                           [Path to grub2-mkconfig]))
+AS_IF([test x$with_grub2_mkconfig_path = x], [
+  dnl Otherwise, look for the path to the system generator.  On some
+  dnl distributions GRUB2 *-mkconfig executable has 'grub2' prefix and
+  dnl on some 'grub'.
+  AC_CHECK_PROGS(GRUB2_MKCONFIG, [grub2-mkconfig grub-mkconfig])
+  dnl Now if we don't find it in the buildroot, hackily search for fedora and
+  dnl dnl use grub2- if found
+  AS_IF([test x$GRUB2_MKCONFIG = x], [
+    AS_IF([grep 'ID_LIKE.*fedora' /etc/os-release], [GRUB2_MKCONFIG=grub2-mkconfig], [GRUB2_MKCONFIG=grub-mkconfig])])
```

Comment:
Yes, anything with a `--with-`, `--without-`, `--enable-` or `--disable-` is easily dealt with in any reasonable distro (particularly if you mention this change in the release notes so I remember that it's changed when it's time to package the next release).

I'd vaguely lean towards defaulting to whatever it's called upstream.